File: nha-trang-bus-tour/public/src/main/java/com/grouppro/nhatrangbustour/Entity/Bus.java
package com.grouppro.nhatrangbustour.Entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity(name = "Bus")
@Table(name = "Bus")
public class Bus implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "BusID", updatable = false)
    private Long BusID;
    @Column(name = "BusNumber", nullable = false)
    private String BusNumber;
    @Column(name = "Seat", nullable = false)
    private int Seat;
    @OneToMany(fetch = FetchType.LAZY)
    @JsonBackReference
    private List<Trip> Trip;
}

File: swd392/lib/model/bus.dart
class Bus {
  int? busId;
  String? busNumber;
  int? seat;

  Bus({this.busId, this.busNumber, this.seat});

  Bus.fromJson(Map<String, dynamic> json) {
    busId = json['busId'];
    busNumber = json['busNumber'];
    seat = json['seat'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['busId'] = this.busId;
    data['busNumber'] = this.busNumber;
    data['seat'] = this.seat;
    return data;
  }
}